src/rpcmessage.rs

```diff
@@ -279,16 +276,7 @@ pub trait RpcMessageMetaTags {
     }
     fn pop_caller_id(&mut self) -> Option<PeerId> {
         let mut ids = self.caller_ids();
-        let id = ids.pop();
-        match id {
-            Some(id) => {
-                self.set_caller_ids(&ids);
-                Some(id)
-            }
-            None => {
-                None
-            }
-        }
+        ids.pop().inspect(|_| { self.set_caller_ids(&ids); })
```

Comment:
`ids.pop()` returns Option<PeerId>, the `inspect()` is called on option just to set_caller_ids, what is not IMO readable oneliner. What about:
```rust
let id = ids.pop();
self.set_caller_ids(&ids);
id
```

Reply:
@fvacek your solution calls set_caller_ids always, inspect only calls its argument on Some

I suggest:
```rust
if let Some(_) = ids.pop() {
  self.set_caller_ids(&ids);
}
```

Reply:
or better yet:
```rust
if ids.pop().is_some() {
  self.set_caller_ids(&ids);
}
```

Reply:
You need `pop()` result to return.
```rust
let id = ids.pop();
if id.is_some() {
    self.set_caller_ids(&ids);
}
id
```
What is exactly the same, what the `inspect()` does, but I like to keep it in this 3 lines longer but IMO more readable form. What I dislike on `inspect` is that the lambda is not using option value, what is not, how the inspect function was designed.

Reply:
>What I dislike on inspect is that the lambda is not using option value, what is not, how the inspect function was designed.

Yeah, I agree, inspect should not be used like that.